src/libcrun/linux.c

```diff
@@ -1684,17 +1687,14 @@ libcrun_create_dev (libcrun_container_t *container, int devfd, int srcfd,
         }
       else
         {
-          char *dirname;
-          cleanup_free char *buffer = NULL;
+          char *dirname = normalized_path, *basename;
           cleanup_close int dirfd = -1;
-          char *basename, *tmp;
-
-          buffer = xstrdup (device->path);
-          dirname = buffer;
-
-          tmp = strrchr (buffer, '/');
-          *tmp = '\0';
-          basename = tmp + 1;
+          // Find the last slash in the normalized path after trimming
+          char *last_slash = strrchr (normalized_path, '/');
+          if (last_slash)
+            basename = last_slash + 1;
+          else
+            basename = normalized_path;
```

Comment:
The previous code looks for the directory name and base directory. We can update it a bit, given the use of `normalized_path`, or keep it at large as-is per the current version.

```c
cleanup_close int dirfd = -1;
cleanup_free char dirname = NULL;
char *basename, *found;

dirname = strdup(normalized_path);

found = strrchr(dirname, '/');
*found = '\0';

basename = found + 1;
```

Reply:
Okay, I think the previous implementation will still work, but the `dirname` points to the same memory as normalized_path and any modifications to `dirname` would also affect normalized_path, which can lead to unintended result if not handled properly. I believe we need a slight modification the code you suggested. Thanks for the pointers.
```c
          cleanup_close int dirfd = -1;
          cleanup_free char *dirname = NULL;
          char *basename, *found;

          dirname = strdup (normalized_path);

          found = strrchr (dirname, '/');
          if (found)
              *found = '\0';

          basename = found + 1;
```

Reply:
@sohankunkerkar, we might have to add a special case, if possible, where the path would be simply a single forward slash (so "/") or any value that is not a path, like "test" (not sure if ever possible to get such a value), then the code would misbehave.

Probalematic values: "/", "test", "test/"; at the moment.